homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,310 @@
+        )
+        for zone_id in hub.zones
+    )
+
+
+def _set_on_commands(command_on_by_id, command_on_dict, hub):
+    for zone_id, zone in hub.zones.items():
+        # Replace unicode non-breaking space (used in Nobø Ecohub) with space
+        zone_name = zone[ATTR_NAME].replace("\xa0", " ")
+        if zone_name not in command_on_dict:
+            continue
+        command_on_name = command_on_dict[zone_name]
+        if not command_on_name:
+            _LOGGER.warning(
+                "Can not turn on (or off) zone '%s', because ON week profile was not specified",
+                zone_name,
+            )
+            continue
+        command_on_id = _get_id_from_name(command_on_name, hub.week_profiles)
+        if not command_on_id:
+            _LOGGER.warning(
+                "Can not turn on (or off) zone '%s', because ON week profile '%s' was not found",
+                zone_name,
+                command_on_name,
+            )
+        else:
+            _LOGGER.debug(
+                "To turn on heater %s '%s', week profile %s '%s' will be used",
+                zone_id,
+                zone_name,
+                command_on_id,
+                command_on_name,
+            )
+            command_on_by_id[zone_id] = command_on_id
+
+
+def _get_id_from_name(name, dictionary):
+    for key in dictionary.keys():
+        # Replace unicode non-breaking space (used in Nobø Ecohub) with space
+        if dictionary[key][ATTR_NAME].replace("\xa0", " ") == name:
+            return key
+    return None
+
+
+class NoboZone(ClimateEntity):
+    """Representation of a Nobø zone.
+
+    A Nobø zone consists of a group of physical devices that are
+    controlled as a unity.
+    """
+
+    _attr_max_temp = MAX_TEMPERATURE
+    _attr_min_temp = MIN_TEMPERATURE
+    _attr_precision = PRECISION_TENTHS
+    _attr_preset_modes = PRESET_MODES
+    # Need to poll to get preset change when in HVACMode.AUTO.
+    _attr_should_poll = True
+    _attr_supported_features = SUPPORT_FLAGS
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(
+        self, zone_id, hub: nobo, command_off_id, command_on_id, override_type
+    ):
+        """Initialize the climate device."""
+        self._id = zone_id
+        self._nobo = hub
+        self._attr_unique_id = hub.hub_serial + ":" + zone_id
+        self._attr_name = hub.zones[self._id][ATTR_NAME]
+        self._attr_hvac_mode = HVACMode.AUTO
+        self._attr_hvac_modes = [HVACMode.HEAT, HVACMode.AUTO]
+        self._command_off_id = command_off_id
+        self._command_on_id = command_on_id
+
+        # Only enable HVACMode.OFF if on- and off-command exists for this zone:
+        if self._command_on_id is not None and self._command_off_id is not None:
+            self._attr_hvac_modes.append(HVACMode.OFF)
+        self._override_type = override_type
+
+        # Register for callbacks before initial update to avoid race condition.
+        hub.register_callback(self._after_update)
```

Comment:
Please register state update callbacks in `async_added_to_hass` entity method.

Reply:
Thanks! Fixed, and I also added a deregister call to `async_will_remove_from_hass`. I assume that was the correct thing to do.